src/planwise/endpoint/projects2.clj

```diff
@@ -34,10 +47,11 @@
       (response project)))
 
    (PUT "/:id" [id project :as request]
-    (println project)
-     (assert (s/valid? ::project project) "invalid project-data")
-     (let [user-id    (util/request-user-id request)]
-       (projects2/update-project service id project)))
+     (let [user-id    (util/request-user-id request)
+           id         (:id project)
+           valid-project    (assoc project :config (apply-default (:config project)))]
```

Comment:
Use the `apply-default` in the `GET "/:id"`, but not in the `PUT "/:id"` route. So the client receives a full config.